core/src/context.rs

```diff
@@ -246,4 +250,59 @@ impl Context {
 
         Ok(())
     }
+
+    pub fn spawn<F>(&self, target: F) where F: 'static + Send + Fn(&mut ContextProxy) {
+        let mut cxp = ContextProxy {
+            current: self.current,
+            event_proxy: self.event_proxy.as_ref().map(|p| p.make_clone())
+        };
+
+        std::thread::spawn(move || target(&mut cxp));
+    }
+}
+
+/// A bundle of data representing a snapshot of the context when a thread was spawned. It supports
+/// a small subset of context operations. You will get one of these passed to you when you create a
+/// new thread with `cx.spawn()`.
+pub struct ContextProxy {
+    pub current: Entity,
+    pub event_proxy: Option<Box<dyn EventProxy>>,
+}
```

Comment:
My only comment for this would be, what else do you think could/should go in here? Are there any properties in `Context` that could be shared atomically between threads? E.g. mouse position or current entity.

Reply:
I looked at the fields and methods on Context, and couldn't find anything that seemed a) useful and b) worth the ergonomics/performance impact of Arc<Mutex<T>>.

Reply:
If they're atomic you don't need the mutex. But yeh I think we'll leave it for now and if at some later date it becomes useful to have more things in `ContextProxy` then we can do that.